### PR TITLE
create db utils

### PR DESCRIPTION
**Pull request by zwwuu**

### CODE REVIEW

File: firebase/utils/blogUtils.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const blogRef = doc(blogCollection);
+  await setDoc(blogRef, { ...data, userId: uid, createdAt: serverTimestamp() });
+};
+
+export const updateBlog = async (uid, data) => {
+  const blogRef = doc(blogCollection, uid);
+  await setDoc(blogRef, data, { merge: true });
+};
+
+export const isBlogExists = async (uid) => {
```
Comment (HamzahMansour) [inline firebase/utils/blogUtils.js:17]: reword, maybe `isExistingBlog` or `doesBlogExist`

File: firebase/utils/userUtils.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const userRef = doc(userCollection, uid);
+  await setDoc(userRef, { ...data, isOnboarded: false, createdAt: serverTimestamp() });
+};
+
+export const updateUser = async (uid, data) => {
+  const userRef = doc(userCollection, uid);
+  await setDoc(userRef, data, { merge: true });
+};
+
+export const isUserExists = async (uid) => {
```
Comment (HamzahMansour) [inline firebase/utils/userUtils.js:17]: same wording as blog ^

**Review by HamzahMansour — DISMISSED** [review]
this is great, some renaming suggestions, but LGTM otherwise, it'll really help with simplifying a lot of the route handlers

File: firebase/utils/blogUtils.js (hunk 1 above)
Comment (thebeatlesphan) [reply firebase/utils/blogUtils.js]: // check if blog exists by id
export const isExistingBlog = async (id) => {
  const blogRef = doc(blogCollection, id);
  const blog = await getDoc(blogRef);

  return blog.exists();
};


function was renamed to 'isExistingBlog'

File: firebase/utils/userUtils.js (hunk 2 above)
Comment (thebeatlesphan) [reply firebase/utils/userUtils.js]: // check if user exists by id
export const isExistingUser = async (id) => {
  const userRef = doc(userCollection, id);
  const user = await getDoc(userRef);

  return user.exists();
};

function renamed to 'isExistingUser'

**Review by thebeatlesphan — APPROVED** [review]
Suggested name changes were made.
Approving and merging.

**Pull request merged**